core/templates/services/math-interactions.service.spec.ts

```diff
@@ -111,25 +111,101 @@ describe('MathInteractionsService', () => {
     expect(mathInteractionsService.validateAlgebraicExpression(
       '(+)', [])).toBeFalse();
     expect(mathInteractionsService.getWarningText()).toBe(
-      'Your answer seems to be missing a variable/number after the "+".');
+      'Your answer seems to be missing a number after the + operator.');
+
+    expect(mathInteractionsService.validateAlgebraicExpression(
+      '*', [])).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'Your answer seems to be missing a number before the × operator.');
+
+    expect(mathInteractionsService.validateAlgebraicExpression(
+      '*a', ['a'])).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'Your answer seems to be missing a number before the × operator.');
+
+    expect(mathInteractionsService.validateAlgebraicExpression(
+      '(*)', [])).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'Your answer seems to be missing a number before the × operator.');
+
+    expect(mathInteractionsService.validateAlgebraicExpression(
+      '(*a)', ['a'])).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'Your answer seems to be missing a number before the × operator.');
+
+    expect(mathInteractionsService.validateAlgebraicExpression(
+      'a*', ['a'])).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'Your answer seems to be missing a number after the × operator.');
+
+    expect(mathInteractionsService.validateAlgebraicExpression(
+      '(a*)', ['a'])).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'Your answer seems to be missing a number after the × operator.');
+
+    expect(mathInteractionsService.validateAlgebraicExpression(
+      '/', [])).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'Your answer seems to be missing a number before the ÷ operator.');
+
+    expect(mathInteractionsService.validateAlgebraicExpression(
+      '/a', ['a'])).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'Your answer seems to be missing a number before the ÷ operator.');
+
+    expect(mathInteractionsService.validateAlgebraicExpression(
+      '(/)', [])).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'Your answer seems to be missing a number before the ÷ operator.');
 
     expect(mathInteractionsService.validateAlgebraicExpression(
-      'a/', ['a'])).toBeFalse();
+      '(/a)', ['a'])).toBeFalse();
     expect(mathInteractionsService.getWarningText()).toBe(
-      'Your answer seems to be missing a variable/number after the "/".');
+      'Your answer seems to be missing a number before the ÷ operator.');
+
+    expect(mathInteractionsService.validateAlgebraicExpression(
+      '-', [])).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'Your answer seems to be missing a number after the - operator.');
```

Comment:
For numeric expressions it's correct that it should be just "number". But for algebraic expressions/equations shouldn't it be "variable/number"?

Reply:
Done!